WcaOnRails/spec/controllers/api_competitions_controller_spec.rb

```diff
@@ -40,4 +40,234 @@
       expect(parsed_body["website"]).to eq "http://example.com"
     end
   end
+
+  describe 'GET #index' do
+    it 'sorts newest to oldest' do
+      yesteryear_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.year.ago)
+      yesterday_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.ago)
+      today_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 0.days.ago)
+      tomorrow_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.from_now)
+
+      get :index
+      expect(response.status).to eq 200
+      json = JSON.parse(response.body)
+      expect(json.map { |c| c["id"] }).to eq [tomorrow_comp, today_comp, yesterday_comp, yesteryear_comp].map(&:id)
```

Comment:
I think the tests need to assert that the produced JSON is the exact string it's supposed to be. I don't know how else we can be sure it follows the documented standard. This kind of thing is extra important for APIs that you'll never interact with as a person and see if something is off.